tools/setup_environment.py

```diff
@@ -34,17 +37,35 @@ def setup_user(user):
     return userdata.pw_uid, userdata.pw_gid
 
 
-def setup_directory(directory, uid, gid):
+def set_group_permissions(gid, path):
+    file_perms = os.stat(path)
+    # add group read permission
+    perms = file_perms.st_mode | stat.S_IRGRP
+    # add group write permission
+    perms = perms | stat.S_IWGRP
+    if os.path.isdir(path):
+        # add group execution permission
+        perms = perms | stat.S_IXGRP
+    os.chmod(path, perms)
+    os.chown(path, -1, gid)
+
+
+def setup_directory(directory, gid):
     """
     Set the directory's owners, creating it if it does not exist.
     """
     utils.create_directory(directory)
-    os.chown(directory, uid, gid)
+    set_group_permissions(gid, directory)
+    for root, dirnames, filenames in os.walk(directory):
+        for f in filenames:
```

Comment:
You could write `for path in dirnames + filenames:` to avoid repeating the for loop.